package-res/resources/web/prompting/builders/ParameterWidgetBuilderBase.js

```diff
@@ -16,14 +16,41 @@
  */
 
 /**
- * The Parameter Widget Builder Base Class
+ * <h2> The Parameter Widget Builder Base Class</h2>
+ *
+ * To use the ParameterWidgetBuilderBase you should require the appropriate file
+ * from Common-Ui:
+ *
+ * <pre><code>
+ *   require(['common-ui/builders/ParameterWidgetBuilderBase'],
+ *     function(ParameterWidgetBuilderBase) {
+ *
+ *     }
+ *   );
+ * </code></pre>
+ *
+ * ParameterWidgetBuilder is an abstract class, that needs to be extended by whoever
+ * needs to create a CDF component. This abstraction is used to create the main properties
+ * needed to build a component
  *
  * @name ParameterWidgetBuilderBase
  * @class
+ * @extends Base
  */
 define(['cdf/lib/Base'], function(Base){
 
   return Base.extend({
+    /**
+     * Assigns to the widget to be built the main cdf properties needed to build a component
+     *
+     * @name ParameterWidgetBuilderBase#build
+     * @method
+     *
+     * @param {Object} args - The object with the properties to build the component
+     * @param {PromptPanel} args.promptPanel - The PromptPanel instance used to get the parameter name, the component name and the html object, where the component will be attached
+     * @param {Parameter} args.param - The parameter with the properties needed to build the component
+     * @returns {{promptType: string, executeAtStart: boolean, param: (args.param|{values, attributes}|{values, attributes, name}|{values, attributes, multiSelect}|{values}|{values, attributes, hasSelection}), name: *, htmlObject: *, type: undefined, parameter: *, postExecution: Function}}
```

Comment:
Is returns statement correct? In generated documentation it's represented as:
Returns:
Type
Object